### ISSUE TITLE
[DOC] `rapids_cython_create_modules()` should document the fact that it reads `CYTHON_FLAGS`

### ISSUE THREAD

**Issue by Jacobfaib** [rapidsai/rapids-cmake#721]
`rapids_cython_create_modules()` inspects the variable `CYTHON_FLAGS`. It's not obvious that this specific variable is consulted. 

This should probably be documented, or perhaps exposed as an additional argument to `rapids_cython_create_modules()`. I guess it is supposed to mimic the `CMAKE_CXX_FLAGS` and friends. But if that is the case, the user might expect to set e.g. `CMAKE_CYTHON_FLAGS` or `CMAKE_CYTHON_FLAGS_DEBUG`.

**Comment by vyasr** [comment]
I agree, this would be good to do. If you have a few moments we'd love a contribution! I originally chose `CYTHON_FLAGS` to match the behavior of [the legacy scikit-build's Cython CMake layer](https://github.com/scikit-build/scikit-build/blob/main/skbuild/resources/cmake/UseCython.cmake#L64), but I'd welcome improvements to that design now that we're pretty much rolling our own Cython invocations. I do hope to eventually switch over to using [cython-cmake](https://github.com/scikit-build/cython-cmake) (and longer-term [a native Cython solution](https://github.com/cython/cython/issues/5486)) so I'd also be happy to propose any such improvements upstream.
